storage/local/series.go

```diff
@@ -459,13 +459,21 @@ func (s *memorySeries) preloadChunksForRange(
 	}
 	if s.chunkDescsOffset != 0 && from.Before(firstChunkDescTime) {
 		cds, err := mss.loadChunkDescs(fp, s.persistWatermark)
+		if s.chunkDescsOffset != -1 && len(cds) != s.chunkDescsOffset {
```

Comment:
Though actually, it would still be good if there's already an error to not overwrite it, since we wouldn't expect the length to match here then anyways?